backend/pkg/notification/sending.go

```diff
@@ -0,0 +1,434 @@
+package notification
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/gobitfly/beaconchain/pkg/commons/db"
+	"github.com/gobitfly/beaconchain/pkg/commons/log"
+	"github.com/gobitfly/beaconchain/pkg/commons/mail"
+	"github.com/gobitfly/beaconchain/pkg/commons/metrics"
+	"github.com/gobitfly/beaconchain/pkg/commons/services"
+	"github.com/gobitfly/beaconchain/pkg/commons/types"
+	"github.com/lib/pq"
+)
+
+func InitNotificationSender() {
+	log.Infof("starting notifications-sender")
+	go notificationSender()
+}
+
+func notificationSender() {
+	for {
+		start := time.Now()
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+
+		conn, err := db.FrontendWriterDB.Conn(ctx)
+		if err != nil {
+			log.Error(err, "error creating connection", 0)
+			cancel()
+			continue
+		}
+
+		_, err = conn.ExecContext(ctx, `SELECT pg_advisory_lock(500)`)
+		if err != nil {
+			log.Error(err, "error getting advisory lock from db", 0)
+
+			err := conn.Close()
+			if err != nil {
+				log.Error(err, "error returning connection to connection pool", 0)
+			}
+			cancel()
+			continue
+		}
+
+		log.Infof("lock obtained")
+		err = dispatchNotifications()
```

Comment:
that won't work as those tx touch a different db. the timeout is here so that the lock eventually gets released. but as the sender has become network specific I believe the lock is also no longer needed or should be applied on network db level in the future.